client/src/components/Dashboard/DashboardForecast/DashboardForecast.tsx

```diff
@@ -1,23 +1,137 @@
 import ButtonSwitch from 'components/ButtonSwitch/ButtonSwitch';
+import Carousel from 'components/Carousel/Carousel';
+import Widget from 'components/Widget/Widget';
 
 // Styles
 import { Flex } from 'styles/global';
 import * as S from '../Dashboard.styles';
 
+interface ForecastTypeState {
+  hourly: boolean;
+  threeHourly: boolean;
+  daily: boolean;
+}
+
 const DashboardForecast: FunctionComponent = () => {
+  const theme = useAppSelector(state => state.theme);
+  const query = useAppSelector(state => state.location.query);
+  const { data: hourlyData, loading: hourlyLoading, error: hourlyError } = useAppSelector(
+    state => state.location.weather.hourly
+  );
+  const {
+    data: threeHourlyData,
+    loading: threeHourlyLoading,
+    error: threeHourlyError
+  } = useAppSelector(state => state.location.weather.threeHourly);
+  const { data: dailyData, loading: dailyLoading, error: dailyError } = useAppSelector(
+    state => state.location.weather.daily
+  );
+
+  const [forecastType, setForecastType] = useState<ForecastTypeState>({
+    hourly: true,
+    threeHourly: false,
+    daily: false
+  });
   const [infoType, setInfoType] = useState<string>('forecast');
+  const [carouselChildPointerEv, setCarouselChildPointerEv] = useState<boolean>(false);
+
+  const dispatch = useDispatch();
 
+  // Handlers
   const handleInfoType = (): void =>
     setInfoType(prevState => (prevState === 'forecast' ? 'air-quality' : 'forecast'));
+  const handleForecastType = (hourly: boolean, threeHourly: boolean, daily: boolean): void =>
+    setForecastType(prevState => ({
+      ...prevState,
+      hourly,
+      threeHourly,
+      daily
+    }));
+  const handleGetLocationHourlyWeather = useCallback(() => {
+    if (query) dispatch(getLocationHourlyWeather(query));
+  }, [dispatch, query]);
+  const handleGetLocationThreeHourlyWeather = useCallback(() => {
+    if (query) dispatch(getLocationThreeHourlyWeather(query));
+  }, [dispatch, query]);
+  const handleGetLocationDailyWeather = useCallback(() => {
+    if (query) dispatch(getLocationDailyWeather(query));
+  }, [dispatch, query]);
+
+  // Get data
+  useEffect(() => {
+    if (forecastType.hourly) handleGetLocationHourlyWeather();
+    if (forecastType.threeHourly) handleGetLocationThreeHourlyWeather();
+    if (forecastType.daily) handleGetLocationDailyWeather();
+  }, [
+    handleGetLocationHourlyWeather,
+    forecastType.hourly,
+    handleGetLocationThreeHourlyWeather,
+    forecastType.threeHourly,
+    handleGetLocationDailyWeather,
+    forecastType.daily
+  ]);
+
+  let data;
+  let loading;
+  let error;
+  if (forecastType.hourly) {
+    data = hourlyData;
+    loading = hourlyLoading;
+    error = hourlyError;
+  }
+  if (forecastType.threeHourly) {
+    data = threeHourlyData;
+    loading = threeHourlyLoading;
+    error = threeHourlyError;
+  }
+  if (forecastType.daily) {
+    data = dailyData;
+    loading = dailyLoading;
+    error = dailyError;
+  }
```

Comment:
too much duplicative code across the whole component. please refactor, choose what request to make, what selector to use, what label to show and other things according to `selectedForecastType` state value